### ISSUE TITLE
"tell" fails silently if a sprite has been renamed

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#2222]
The following worked in a fresh Snap!

![image](https://user-images.githubusercontent.com/2138696/46486280-531b1980-c7f5-11e8-8024-b721c2b21605.png)

until "Sprite" was renamed to something else.

**Comment by jguille2** [comment]
Yes @ToonTalk , but I don't know if this is an issue... or it is the expected behavior.

What you have related is happening everywhere we have the sprite name as an input: `tell`, `ask`, `point towards`, `go to`...
To fix it, we'll need something like "rename/rename all" in vars.

But we can play with our first class features... leaving sprite names, and referencing objects itself.
![spritesarefirstclass](https://user-images.githubusercontent.com/2926315/46488581-5cab7e00-c803-11e8-8639-db2572c905e0.png)

**Comment by ToonTalk** [comment]
I agree the first class way of avoiding names is nicer.

I'm not sure if rename all is warranted but at least some kind of feedback if a tell, ask, point, ... receives the name of a non-existent sprite.

**Comment by towerofnix** [comment]
Note that in Scratch, everything is automatically renamed. If you change the name of a sprite, all references (in dropdowns) to it are automatically updated; same deal for sounds and costumes.

**Comment by brianharvey** [comment]
I agree, if a sprite is renamed, any dropdown reference to it should also automatically be renamed.  If we had implemented first class sprites from the beginning, we wouldn't even _have_ references to sprite names in primitive blocks, except that we'd have a reporter named OBJECT with sprite names as inputs reporting a sprite.  Oh, and I just checked, and sure enough BYOB3's OBJECT block's name input does get renamed if the sprite is renamed.  So that's the right behavior.  :-)

**Comment by DyslexicAwe** [comment]
Voting for this.
